File: backend/app/schemas/exchange_rates.py
from datetime import datetime

from pydantic import BaseModel


class ExchangeRateSnapshot(BaseModel):
    base_currency: str = "USD"
    rates: dict[str, float]
    updated_at: datetime
    next_update_at: datetime
    provider: str
    provider_url: str
    update_frequency: str
    stale: bool = False
    refresh_after_seconds: int = 60

File: frontend/lib/api/exchange-rates.ts
import { api } from "./client";

export interface ExchangeRateSnapshot {
  base_currency: string;
  rates: Record<string, number>;
  updated_at: string;
  next_update_at: string;
  provider: string;
  provider_url: string;
  update_frequency: string;
  stale: boolean;
  refresh_after_seconds: number;
}

export async function fetchExchangeRates(signal: AbortSignal): Promise<ExchangeRateSnapshot> {
  const response = await api.get<ExchangeRateSnapshot>("/exchange-rates/latest", { signal, timeout: 12000 });
  return response.data;
}
